nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -326,11 +329,8 @@ def training_step(self, dataloader_iter, batch_idx):
             model=[self.model],
             num_microbatches=get_num_microbatches(),
             forward_only=False,
-            tensor_shape=tensor_shape,
-            dtype=self.autocast_dtype,
-            grad_scaler=self.trainer.precision_plugin.scaler.scale if self.cfg.precision == 16 else None,
-            sequence_parallel=self.cfg.get('sequence_parallel', False),
-            enable_autocast=self.enable_autocast,
+            seq_length=self.cfg.encoder_seq_length,
```

Comment:
looks like we were using different seq lens based on dataloader type, need to update that here
Also, we can remove the tenasor_shape variable declaration code above

Reply:
Nice find. Fixed.